pkg/tinyvcdiff/contrib/tinyvcdiff_vfs/vfs.c

```diff
@@ -0,0 +1,66 @@
+#include "vcdiff_vfs.h"
+#include "errno.h"
+
+static int read (void *dev, uint8_t *dest, size_t offset, size_t len)
+{
+    int rc;
+    vcdiff_vfs_t *vfs = (vcdiff_vfs_t *) dev;
+
+    /* seek to the given offset */
+    rc = vfs_lseek(vfs->fd, offset, SEEK_SET);
+    if (rc < 0) {
+        return rc;
+    } else if ((size_t) rc != offset) {
+        return -EFAULT;
+    }
+
+    /* write until all bytes are stored in the buffer */
+    offset = 0;
+    while (len) {
+        rc = vfs_read(vfs->fd, &dest[offset], len);
+        if (rc < 0) {
+            return rc;
+        }
+        len -= rc;
+        offset += rc;
```

Comment:
Shouldn't it be?
```suggestion
        rc = vfs_read(vfs->fd, dest, len);
        if (rc < 0) {
            return rc;
        }
        len -= rc;
        dest += rc;
```

Or is the `offset` really *both* into the destination buffer *and* the source file?

Reply:
No, just the offset in the file, not the buffer. I'm reusing the variable and set it to `0` before entering the loop.

But I guess moving the pointer is also a common way to go that is understood by most programmers ;-) Changed it.

50a48a8a489417608b095e44c198e9b331c574fd